framework/instance_gen.sage

```diff
@@ -111,28 +114,34 @@ def initialize_LAC_instance(dbdd_class, n, q, m, D_e, D_s, verbosity=1):
     B = kannan_embedding(B, tar)
     u = concatenate([e, s, [1]])
     # define the mean and sigma of the instance
-    mu_e, s_e = average_variance(D_e)
-    mu_s, s_s = average_variance(D_s)
+    mu_e, s_e = 0, 1./2
+    mu_s, s_s = 0, 1./2
     mu = vec(m * [mu_e] + n * [mu_s] + [1])
     S = diagonal_matrix(m * [s_e] + n * [s_s] + [0])
     return A, b, dbdd_class(B, S, mu, u, verbosity=verbosity)
 
 
-def initialize_NTRU_instance(dbdd_class, n, q, D_f, D_g, verbosity=1):
+def initialize_NTRU_instance(dbdd_class, n, q, Df, Dg, verbosity=1):
     if verbosity:
-        logging("     Build DBDD for NTRU HPS VERSION     ", style="HEADER")
+        logging("     Build DBDD from an NTRU instance (h=f/g [q])  ", style="HEADER")
         logging("n=%3d \t \t q=%d" % (n, q), style="VALUE")
 
-    assert (q % 16 == 0), "NTRU-HPS requires 16 to divide q"
-
-    # This construction of the lattice is not correct for NTRU
-    # It doesn't matter when we are only doing prediction though
-    if dbdd_class==DBDD:
-        raise NotImplementedError("NTRU KeyGen not actually implemented.")
-
-    B = diagonal_matrix(n * [q] + n * [1])
-    mu_f, s_f = average_variance(D_f)
-    mu_g, s_g = average_variance(D_g)
+    mu_f, s_f = QQ(0), QQ(2*Df/n)
+    mu_g, s_g = QQ(0), QQ(2*Dg/n)
     mu = vec(n * [mu_f] + n * [mu_g])
     S = diagonal_matrix(n * [s_f] + n * [s_g])
-    return None, None, dbdd_class(B, S, mu, None, homogeneous=True, verbosity=verbosity)
+
+
+    if dbdd_class not in [DBDD, DBDD_optimized]:
+        B = diagonal_matrix(n * [q] + n * [1])
```

Comment:
If the idea of creating a fake basis is to give the volume for DBDD_predict(_diag), it can be directly done thanks to the parameters "Bvol". No need to create this B.